### PR TITLE
[IMP] sale, project: Invoicing flow improvements

### PR DESCRIPTION
**Pull request by damrOdoo**
This commit's purpose is to ease the the flow of invoicing from the project app. The warning message when the user tries to invoice without fulfilling the condition has been updated.
When the customer of an SOL is a person and belongs to a company, the company is now also shown on the sol, in addition of the name of the customer. The idea is that the user is more keen to recognize the company than just the name of the person.

task-3251615

### CODE REVIEW

**Comment by Feyensv** [comment]
Please create and keep your PR's in draft state until they have been validated and tested in your team ;) (i.e. the task is in Testing BE/Review).  Codeowners should review the final version of a task, not the work in progress.

File: addons/sale_service/models/sale_order_line.py
Comment (MissingNoShiny) [inline addons/sale_service/models/sale_order_line.py:40]: I don't think it makes sense to change this just in some views, that's inconsistent and confusing. That'll be something to see with AJU and DALA.

File: addons/project/views/project_task_views.xml
Comment (MissingNoShiny) [inline addons/project/views/project_task_views.xml:150]: If we do end up going this route however, we should add the context key in other places, because as is, it won't be active if you go through "My Tasks" or "All Tasks", or even when accessing the task directly through the link

**Review by MissingNoShiny — COMMENTED** [review]
Thanks for your work, I left some comments :slightly_smiling_face:

File: addons/sale_service/models/sale_order_line.py
Comment (damrOdoo) [reply addons/sale_service/models/sale_order_line.py]: they've discussed about it, and agreed to simply add the change to the generic name get of sols, not just for task views.

**Review by auon-odoo — APPROVED** [review]
Thanks! :)

File: addons/sale/models/sale_order_line.py
Comment (xavierbol) [inline addons/sale/models/sale_order_line.py:1162]: I think we should instead take the ancestor instead of just the parent because the parent could be the manager of the customer and not his company for instance. 🤔

**Review by xavierbol — COMMENTED** [review]
Thanks for your work, I left one comment and after it will look good to me. 🙂

File: addons/sale/models/sale_order_line.py
Comment (damrOdoo) [reply addons/sale/models/sale_order_line.py:1164]: It looks like it is. Thanks for your review :smile:

File: addons/sale/models/sale_order_line.py
Comment (Feyensv) [inline addons/sale/models/sale_order_line.py:1]: :eyes:

Comment (damrOdoo) [reply addons/sale/models/sale_order_line.py]: :sweat_smile:

**Review by MissingNoShiny — APPROVED** [review]

**Review by xavierbol — APPROVED** [review]
LGTM, thanks for your work! 🙂 

Hello @Feyensv , thanks for your review 🙂 , everything is okay for you? 🙂 

Thanks in advance! 

Have a nice day! 🙂

**Review by xavierbol — COMMENTED** [review]
@damrOdoo can you change the commit title to just mention sale module? 🙂 

Thanks in advance! 🙂

File: addons/sale/models/sale_order_line.py
Comment (vava-odoo) [inline addons/sale/models/sale_order_line.py:1185]: Shouldn't this conditional line be changed as well? It made sense before, using the `order_partner_id`, but now, it looks strange to me.

**Review by Feyensv — APPROVED** [review]
Valentin's remark makes sense, the if line can be dropped IMHO (I doubt partners without name are really frequent anyway, if possible).

For the rest, not so big changes, it's ok for me (but poor translators with the nothing to invoice error message that changes every year :D)

**Comment by xavierbol** [comment]
robodoo r+

**Pull request closed without merging**
